cmd/helm/repo_list.go

```diff
@@ -97,13 +97,39 @@ func (r *repoListWriter) encodeByFormat(out io.Writer, format output.Format) err
 	return nil
 }
 
+// Returns all repos from repos, except those with names matching ignoredRepoNames
+// Inspired by https://stackoverflow.com/a/28701031/893211
+func filterRepos(repos []*repo.Entry, ignoredRepoNames []string) []*repo.Entry {
+	// if ignoredRepoNames is nil, just return repo
+	if ignoredRepoNames == nil {
+		return repos
+	}
+
+	filteredRepos := make([]*repo.Entry, 0)
+
+	ignored := make(map[string]bool, 0)
+	for _, repo := range ignoredRepoNames {
```

Comment:
Nit: to avoid using the same variable name with two different types, can you rename `repo` here to `name` or `repoName`?